components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java

```diff
@@ -500,4 +600,156 @@ public void testSetCookieExistCookieConfig() {
         assertEquals(storedCookie.getMaxAge(), age);
     }
 
+    @Test
+    public void testSetCookieWithSameSiteCookieNonExistCookieConfig() {
+
+        mockCookieTest();
+        int age = 3600;
+
+        FrameworkUtils.setCookie(request, response, FrameworkConstants.COMMONAUTH_COOKIE, "commonAuthIdValue", age, SameSiteCookie.STRICT, "Dummy-Path");
+
+        verify(response, times(1)).addCookie(cookieCaptor.capture());
+        List<Cookie> capturedCookies = cookieCaptor.getAllValues();
+        Cookie storedCookie = capturedCookies.get(0);
+
+        assertEquals(storedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
+        assertEquals(storedCookie.getPath(), "Dummy-Path");
+        assertEquals(storedCookie.getMaxAge(), age);
+
+    }
+
+    @Test
+    public void testSetCookieWithSameSiteExistCookieConfig() {
+
+        mockCookieTest();
+        IdentityCookieConfig cookieConfig = new IdentityCookieConfig(FrameworkConstants.COMMONAUTH_COOKIE);
+        IdentityUtil.getIdentityCookiesConfigurationHolder().put(FrameworkConstants.COMMONAUTH_COOKIE, cookieConfig);
+        int age = 3600;
+
+        FrameworkUtils.setCookie(request, response, FrameworkConstants.COMMONAUTH_COOKIE, "commonAuthIdValue", age, SameSiteCookie.STRICT, "Dummy-Path");
+
+        verify(response, times(1)).addCookie(cookieCaptor.capture());
+        List<Cookie> capturedCookies = cookieCaptor.getAllValues();
+        Cookie storedCookie = capturedCookies.get(0);
+
+        assertEquals(storedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
+        assertEquals(storedCookie.getPath(), "Dummy-Path");
+        assertEquals(storedCookie.getMaxAge(), age);
+    }
+
+    @Test
+    public void testGetCookieExistingCookie() {
+
+        Cookie[] cookies = getAuthenticationCookies();
+        when(request.getCookies()).thenReturn(cookies);
+
+        Cookie out = FrameworkUtils.getCookie(request, cookies[0].getName());
+
+        assertEquals(out, cookies[0]);
+    }
+
+    @Test
+    public void testGetCookieNonExistingCookie() {
+
+        Cookie out = FrameworkUtils.getCookie(request, "nonExistingCookie");
+        assertNull(out);
+    }
+
+    @Test
+    public void testGetHashOfCookie() {
+
+        Cookie cookie = new Cookie(FrameworkConstants.COMMONAUTH_COOKIE, "commonAuthIdValue");
+
+        String out = FrameworkUtils.getHashOfCookie(cookie);
+
+        assertEquals(out, DigestUtils.sha256Hex("commonAuthIdValue"));
```

Comment:
fix formatting

Reply:
Addressed with commit 387514e50f42aead307ee0f529bb17b806672a14
(fixed suggested and some other formatting issue)